File: sdks/android/FraudDetection.kt
package com.example.fraudsdk

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.MediaType.Companion.toMediaType
import okhttp3.RequestBody.Companion.toRequestBody
import org.json.JSONObject
import java.io.IOException

class FraudClient(
    private val apiKey: String,
    private val baseURL: String = "https://api.example.com/v1"
) {
    private val client = OkHttpClient.Builder()
        .addInterceptor { chain ->
            val original = chain.request()
            val request = original.newBuilder()
                .header("Authorization", "Bearer $apiKey")
                .header("Content-Type", "application/json")
                .header("X-Platform", "android")
                .header("X-SDK-Version", "1.0.0")
                .header("X-SDK-Type", "android-native")
                .method(original.method, original.body)
                .build()
            chain.proceed(request)
        }
        .build()

    @Throws(IOException::class)
    suspend fun getMetrics(): MetricsResponse = withContext(Dispatchers.IO) {
        val request = Request.Builder()
            .url("$baseURL/metrics")
            .get()
            .build()

        client.newCall(request).execute().use { response ->
            if (!response.isSuccessful) throw IOException("Unexpected code $response")
            
            val body = response.body?.string() ?: throw IOException("Empty response")
            val json = JSONObject(body)
            
            MetricsResponse(
                riskScore = json.getDouble("riskScore"),
                activeUsers = json.getInt("activeUsers"),
                alertCount = json.getInt("alertCount")
            )
        }
    }

    @Throws(IOException::class)
    suspend fun trackInteraction(data: JSONObject): InteractionResponse = withContext(Dispatchers.IO) {
        val mediaType = "application/json; charset=utf-8".toMediaType()
        val body = data.toString().toRequestBody(mediaType)
        
        val request = Request.Builder()
            .url("$baseURL/track")
            .post(body)
            .build()

        client.newCall(request).execute().use { response ->
            if (!response.isSuccessful) throw IOException("Unexpected code $response")
            
            val responseBody = response.body?.string() ?: throw IOException("Empty response")
            val json = JSONObject(responseBody)
            
            InteractionResponse(
                success = json.getBoolean("success"),
                transactionId = json.optString("transactionId")
            )
        }
    }
}

data class MetricsResponse(
    val riskScore: Double,
    val activeUsers: Int,
    val alertCount: Int
)

data class InteractionResponse(
    val success: Boolean,
    val transactionId: String?
)

File: sdks/go/sdks/go/sdks/ios/FraudDetection.swift
import Foundation

public class FraudClient {
    private let baseURL: String
    private let apiKey: String
    private let session: URLSession
    
    public init(apiKey: String, baseURL: String = "https://udene.net/v1") {
        self.apiKey = apiKey
        self.baseURL = baseURL
        
        let config = URLSessionConfiguration.default
        config.httpAdditionalHeaders = [
            "Authorization": "Bearer \(apiKey)",
            "Content-Type": "application/json",
            "X-Platform": "ios",
            "X-SDK-Version": "1.0.0",
            "X-SDK-Type": "ios-native"
        ]
        self.session = URLSession(configuration: config)
    }
    
    public func getMetrics(completion: @escaping (Result<MetricsResponse, Error>) -> Void) {
        guard let url = URL(string: "\(baseURL)/metrics") else {
            completion(.failure(FraudError.invalidURL))
            return
        }
        
        let task = session.dataTask(with: url) { data, response, error in
            if let error = error {
                completion(.failure(error))
                return
            }
            
            guard let data = data else {
                completion(.failure(FraudError.noData))
                return
            }
            
            do {
                let metrics = try JSONDecoder().decode(MetricsResponse.self, from: data)
                completion(.success(metrics))
            } catch {
                completion(.failure(error))
            }
        }
        task.resume()
    }
    
    public func trackInteraction(_ data: [String: Any], completion: @escaping (Result<InteractionResponse, Error>) -> Void) {
        guard let url = URL(string: "\(baseURL)/track") else {
            completion(.failure(FraudError.invalidURL))
            return
        }
        
        var request = URLRequest(url: url)
        request.httpMethod = "POST"
        
        do {
            request.httpBody = try JSONSerialization.data(withJSONObject: data)
        } catch {
            completion(.failure(error))
            return
        }
        
        let task = session.dataTask(with: request) { data, response, error in
            if let error = error {
                completion(.failure(error))
                return
            }
            
            guard let responseData = data else {
                completion(.failure(FraudError.noData))
                return
            }
            
            do {
                let response = try JSONDecoder().decode(InteractionResponse.self, from: responseData)
                completion(.success(response))
            } catch {
                completion(.failure(error))
            }
        }
        task.resume();
    }
}

public enum FraudError: Error {
    case invalidURL
    case noData
    case decodingError
}

public struct MetricsResponse: Codable {
    public let riskScore: Double
    public let activeUsers: Int
    public let alertCount: Int
}

public struct InteractionResponse: Codable {
    public let success: Bool
    public let transactionId: String?
}
